src/fread.c

```diff
@@ -66,6 +66,9 @@ static size_t fileSize;
 static int8_t *type = NULL, *tmpType = NULL, *size = NULL;
 static lenOff *colNames = NULL;
 static freadMainArgs args = {0};  // global for use by DTPRINT; static implies ={0} but include the ={0} anyway just in case for valgrind #4639
+#ifdef __EMSCRIPTEN__
+  static int mmp_fd = -1; // Under Wasm we must keep the fd open until we're done with the mmap, due to an Emscripten bug (#5969).
```

Comment:
```suggestion
  // Under Wasm we must keep the fd open until we're done with the mmap, due to an Emscripten bug (#5969).
  // TODO(emscripten-core/emscripten#20459): revert this
  static int mmp_fd = -1;
```

BTW, what does this mean in terms of dependencies? I.e., if/when the Emscripten bug is fixed, if we remove this patch, it induces a version dependency on Emscripten, right? What would be the proper way to express that? Something within webR only? Something in our DESCRIPTION file?

Reply:
> if we remove this patch, it induces a version dependency on Emscripten

Technically true, but in any case packages built for a specific version of webR already strictly depend on a specific version of Emscripten, since Emscripten does not guarantee ABI compatibility between versions.

I think such a strong dependency link should probably be handled by webR itself and the [rwasm](https://github.com/r-wasm/rwasm) package, since it will be the same for all R packages built for a certain version of webR anyway.